lib/admin.php

```diff
@@ -227,6 +227,23 @@ function wc_admin_enqueue_script() {
 }
 add_action( 'admin_enqueue_scripts', 'wc_admin_enqueue_script' );
 
+/**
+ * Hydrates the initial state.
+ */
+function wc_admin_hydrate_state() {
+	$current_user_data = array();
+	foreach ( wc_admin_get_user_data_fields() as $user_field ) {
+		$current_user_data[ $user_field ] = get_user_meta( get_current_user_id(), 'wc_admin_' . $user_field, true );
+	}
+
+	wp_localize_script(
+		WC_ADMIN_APP,
+		'wcAdminServer',
+		array( 'currentUserData' => $current_user_data )
+	);
+}
+add_action( 'admin_enqueue_scripts', 'wc_admin_hydrate_state' );
```

Comment:
What do you think about adding `currentUserData` to the `wcSettings` object we're already creating? https://github.com/woocommerce/wc-admin/blob/eaffb0d607f48c3999f2a0dc834e0ea98d82f984/lib/client-assets.php#L139

Reply:
I like the idea of consolidating where this code gets placed, but I'm also a little wary of adding this to `wcSettings` since I believe that's traditionally used for global settings and user specific settings are kept separate.

It looks like we're already some user specific settings to the `wcSettings` global, so I'll append it there for now and we can re-address later if we need to separate.